cleanlab/multiannotator.py

```diff
@@ -0,0 +1,436 @@
+import warnings
+
+# TODO: unsure where to place this function? should it go to cleanlab.rank or stay here
+def _get_label_quality_scores_with_NA(
+    labels: pd.Series,
+    pred_probs: np.array,
+    kwargs: dict = {},
+) -> pd.Series:
+    label_quality_scores_subset = get_label_quality_scores(
+        labels=labels[pd.notna(labels)],
+        pred_probs=pred_probs[pd.notna(labels)],
+        **kwargs,
+    )
+    label_quality_scores = pd.Series(np.nan, index=np.arange(len(labels)))
+    label_quality_scores[pd.notna(labels)] = label_quality_scores_subset
+    return label_quality_scores
+
+
+def get_consensus_labels(
+    labels_multiannotator: pd.DataFrame,
+    pred_probs: np.array,
+    method: str,
+) -> np.array:
+    """Returns consensus labels aggregated from multiple annotators, determined by method specified.
+
+    Parameters
+    ----------
+    labels_multiannotator : pd.DataFrame
+        2D pandas DataFrame of multiple given labels for each example with shape (N, M),
+        where N is the number of examples and M is the number of annotators.
+        For more details, labels in the same format expected by the :py:func:`get_label_quality_scores_multiannotator <cleanlab.multiannotator.get_label_quality_scores_multiannotator>`.
+
+    pred_probs : np.array
+        An array of shape ``(N, K)`` of model-predicted probabilities, ``P(label=k|x)``.
+        For details, predicted probabilities in the same format expected by the :py:func:`get_label_quality_scores_multiannotator <cleanlab.multiannotator.get_label_quality_scores_multiannotator>`.
+
+    method : str (Options: ["majority"])
+        The method used to aggregate labels from multiple annotators into a single consensus label. Default is by using the majority vote.
+        Options include:
+        - ``"majority"``: means consensus labels are reached via a simple majority vote among annotators, with ties broken via ``pred_probs``.
+
+    Returns
+    -------
+    consensus_labels: np.array
+        An array of shape ``(N,)`` with the consensus labels aggregated from all annotators.
+    """
+
+    valid_methods = ["majority"]
+    consensus_labels = np.empty(len(labels_multiannotator), dtype="int32")
+
+    # TODO: is there a more efficient/elegant way to do this?
+    if method == "majority":
+        mode_labels_multiannotator = labels_multiannotator.mode(axis=1)
+        for i in range(len(mode_labels_multiannotator)):
+            consensus_labels[i] = mode_labels_multiannotator.iloc[i][
+                pred_probs[
+                    i, mode_labels_multiannotator.iloc[i].dropna().astype(int).to_numpy()
+                ].argmax()
+            ]
+    else:
+        raise ValueError(
+            f"""
+            {method} is not a valid consensus method!
+            Please choose a valid consensus_method: {valid_methods}
+            """
+        )
+
+    return consensus_labels
+
+
+def get_annotator_agreement(
```

Comment:
Let's make this helper method (with underscore prefix): `_get_annotator_agreement()`

So we don't have to commit to supporting this exact API forever.